File: 0011-盛最多水的容器/Solution.py
# 双指针
class Solution:
    def maxArea(self, height):
        low, high = 0, len(height)-1
        res = 0
        while(low < high):
            temp = 0
            if height[low] <= height[high]:
                temp = height[low]
                low = low + 1
            else:
                temp = height[high]
                high = high - 1
            
            temp = (high-low + 1) * temp
            if temp > res:
                res = temp
        return res

File: 0011-盛最多水的容器/Solution.java
class Solution {
    public int maxArea(int[] height) {
        int res = 0, temp = 0;
        for (int i = 0; i < height.length-1; i++) {
            for(int j = i+1; j<height.length; j++){
                temp = (j - i) * (height[i]<height[j]? height[i]: height[j]);
                res = temp > res? temp: res;
            }
        }
        return res;
    }
}